src/ucp/core/ucp_worker.c

```diff
@@ -2171,6 +2180,34 @@ ucs_status_t ucp_worker_create(ucp_context_h context,
         ucs_strided_alloc_init(&worker->ep_alloc, sizeof(ucp_ep_t), 2);
     }
 
+    if (params->field_mask & UCP_WORKER_PARAM_FIELD_USR_MEM_ALLOC) {
+
+        if (params->user_mem_allocator_init == NULL ||
+            params->user_mem_allocator_malloc == NULL ||
+            params->user_mem_allocator_free == NULL) {
+
+            ucs_error("Missing user allocator params");
+            status = UCS_ERR_INVALID_PARAM;
+            goto err_free;
+        }
+
+        worker->user_allocator.ops.init = params->user_mem_allocator_init;
+        worker->user_allocator.ops.cleanup = params->user_mem_allocator_cleanup;
+        worker->user_allocator.ops.malloc = params->user_mem_allocator_malloc;
+        worker->user_allocator.ops.free = params->user_mem_allocator_free;
+        worker->user_allocator.malloc_cb = ucp_worker_usr_allocator_malloc_cb;
```

Comment:
Maybe remove `ucp_worker_usr_allocator_malloc_cb`, and for the purpose of this API PR, make a deeper documentation of the `malloc_cb` member instead.